src/libostree/ostree-repo-pull.c

```diff
@@ -2066,14 +2076,7 @@ ostree_repo_pull_with_options (OstreeRepo             *self,
                                             &configured_branches, error))
     goto out;
 
-  if (strcmp (soup_uri_get_scheme (pull_data->base_uri), "file") == 0)
-    {
-      g_autoptr(GFile) remote_repo_path = g_file_new_for_path (soup_uri_get_path (pull_data->base_uri));
-      pull_data->remote_repo_local = ostree_repo_new (remote_repo_path);
-      if (!ostree_repo_open (pull_data->remote_repo_local, cancellable, error))
-        goto out;
-    }
-  else
+  if (!pull_data->remote_repo_local)
```

Comment:
These two are not the same.
The old code checked if the url (likely taken from the remote config) starts with file:
However, remote_repo_local is only set above if you pass in a direct url rather than a remote name.